controlplane/api/v1beta1/conversion.go

```diff
@@ -72,25 +73,41 @@ func Convert_v1beta2_KThreesConfigSpec_To_v1beta1_KThreesConfigSpec(in *bootstra
 
 // ConvertTo converts the v1beta1 KThreesControlPlane receiver to a v1beta2 KThreesControlPlane.
 func (in *KThreesControlPlane) ConvertTo(dstRaw ctrlconversion.Hub) error {
-	dst := dstRaw.(*cabp3v1.KThreesControlPlane)
+	dst := dstRaw.(*controlplanev1beta2.KThreesControlPlane)
 	if err := Convert_v1beta1_KThreesControlPlane_To_v1beta2_KThreesControlPlane(in, dst, nil); err != nil {
```

Comment:
Do we need similar conversion logic to Convert_v1beta1_KThreesControlPlane_To_v1beta2_KThreesControlPlane?
```
if !in.DisableExternalCloudProvider { out.CloudProviderName = "external" out.DisableCloudController = true } else { out.CloudProviderName = "" out.DisableCloudController = false }
```

Reply:
I think in that case that mapping the deprecated field should be enough, otherwise we are going to overwrite this information with any conversion logic we add. 

Still I am not sure, because you could also argue that the deprecated field should **not** be used in v1beta2, therefore we could ignore it for conversion.

So either we leave it like it is, or I can implement something like:

```go
	if in.CloudProviderName != "external" &&  !in.DisableCloudController {
		out.DisableExternalCloudProvider = true
	} else {
		out.DisableExternalCloudProvider = false
	}
```

This _should_ work I believe, even though it's confusing and I wonder about corner cases.
Do you have a preference?

Reply:
cool~ thanks